src/main/kotlin/com/yapp/giljob/global/util/S3Util.kt

```diff
@@ -0,0 +1,45 @@
+package com.yapp.giljob.global.util
+
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.s3.model.ObjectMetadata
+import com.amazonaws.services.s3.model.S3Object
+import com.yapp.giljob.global.error.ErrorCode
+import com.yapp.giljob.global.error.exception.BusinessException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.mock.web.MockMultipartFile
+import org.springframework.stereotype.Component
+import org.springframework.web.multipart.MultipartFile
+import java.util.*
+
+@Component
+class S3Util {
+    private val log : Logger get() = LoggerFactory.getLogger(this::class.java)
+
+    @Autowired
+    private lateinit var amazonS3: AmazonS3
+
+    @Value("\${cloud.aws.s3.bucket}")
+    private lateinit var bucketName: String
+
+    fun fileUplaod(file: MultipartFile): String {
+        var fileName = ""
+        try {
+            fileName = UUID.randomUUID().toString() + file.originalFilename
+            val metadata = ObjectMetadata()
+            metadata.contentLength = file.size
+            amazonS3.putObject(bucketName, fileName, file.inputStream, metadata)
+        } catch (e: Exception) {
+            throw BusinessException(ErrorCode.FILE_UPLOAD_ERROR)
+        }
+        log.info("File Uploaded Successfully : {}", fileName)
+        return fileName
+    }
```

Comment:
```suggestion
fun fileUplaod(file: MultipartFile): String {
    val fileName: String
    runCatching {
        fileName = UUID.randomUUID().toString() + file.originalFilename ?: ""
        val metadata = ObjectMetadata()
        metadata.contentLength = file.size
    }.onSuccess {
        amazonS3.putObject(bucketName, fileName, file.inputStream, metadata)
    }.onFailure {
        throw BusinessException(ErrorCode.FILE_UPLOAD_ERROR)
    }
    log.info("File Uploaded Successfully : {}", fileName)
    return fileName
}
```

참고차.. 코틀린스럽게 try - catch 짜보았습니다 .. 코틀린에서는 runCatching 로도 자주 사용돼요!

Reply:
오...! 어쩐지 코틀린에서는 try catch가 별로 없는 것 같아서 찾아보긴 했었는데..! 이렇게 다시 보니까 전에 짠 코드를 너무 이상하게 짰네욬ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ runCatching,,메모,,,